foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java

```diff
@@ -128,122 +159,309 @@ void testErrorHandling() {
-            getClass()
-                .getClassLoader()
-                .getResourceAsStream("embeddingResponse.json")
-                .readAllBytes());
-    stubFor(post(anyUrl()).willReturn(okJson(response)));
-
-    final var request = new OpenAiEmbeddingParameters().setInput("Hello World");
-    final var result = client.embedding(request);
-
-    assertThat(result).isNotNull();
-    assertThat(result.getModel()).isEqualTo("ada");
-    assertThat(result.getObject()).isEqualTo("list");
-
-    assertThat(result.getUsage()).isNotNull();
-    assertThat(result.getUsage().getCompletionTokens()).isNull();
-    assertThat(result.getUsage().getPromptTokens()).isEqualTo(2);
-    assertThat(result.getUsage().getTotalTokens()).isEqualTo(2);
-
-    assertThat(result.getData()).isNotNull().hasSize(1);
-    var embeddingData = result.getData().get(0);
-    assertThat(embeddingData).isNotNull();
-    assertThat(embeddingData.getObject()).isEqualTo("embedding");
-    assertThat(embeddingData.getIndex()).isEqualTo(0);
-    assertThat(embeddingData.getEmbedding())
-        .isNotNull()
-        .isNotEmpty()
-        .containsExactly(-0.0000000070958645d, 2.123e-300d, -0.0069813123d, -3.385849e-05d)
-        .hasToString(
-            "[-7.0958645E-9, 2.123E-300, -0.0069813123, -3.385849E-5]"); // ensure double precision
-
-    verify(
-        postRequestedFor(urlPathEqualTo("/embeddings"))
-            .withRequestBody(
-                equalToJson("""
-                                {"input":["Hello World"]}""")));
+  void streamChatCompletion() throws IOException {
+    try (var inputStream =
+        spy(
+            Objects.requireNonNull(
+                getClass().getClassLoader().getResourceAsStream("streamChatCompletion.txt")))) {
+
+      final var httpClient = mock(HttpClient.class);
+      ApacheHttpClient5Accessor.setHttpClientFactory(destination -> httpClient);
+
+      // Create a mock response
+      final var mockResponse = new BasicClassicHttpResponse(200, "OK");
+      final var inputStreamEntity = new InputStreamEntity(inputStream, ContentType.TEXT_PLAIN);
+      mockResponse.setEntity(inputStreamEntity);
+      mockResponse.setHeader("Content-Type", "text/event-stream");
+
+      // Configure the HttpClient mock to return the mock response
+      doReturn(mockResponse).when(httpClient).executeOpen(any(), any(), any());
+
+      final var request =
+          new OpenAiChatCompletionParameters()
+              .setMessages(
+                  List.of(
+                      new OpenAiChatUserMessage()
+                          .addText(
+                              "Can you give me the first 100 numbers of the Fibonacci sequence?")));
+
+      try (Stream<OpenAiChatCompletionDelta> stream = client.streamChatCompletion(request)) {
+
+        OpenAiChatCompletionOutput totalOutput = new OpenAiChatCompletionOutput();
+        final List<OpenAiChatCompletionDelta> deltaList =
+            stream.peek(totalOutput::addDelta).toList();
```

Comment:
To assert the stream isn't consumed prematurely I'd try something like:

```suggestion
final List<OpenAiChatCompletionDelta> deltaList =
            stream.peek(totalOutput::addDelta)
                    .peek(any -> {
                      try {
                        assertThat(inputStream.available() > 0);
                      } catch (IOException e) {
                        throw new RuntimeException(e);
                      }
                    })
                    .toList();

        assertThat(inputStream.available()).isZero();
```

The IO exception is annoying, maybe moving the assertion to a small helper method so that it looks better.